src/UuidGenerator.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * This file is part of the ramsey/uuid-doctrine library
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright Copyright (c) Ben Ramsey <http://benramsey.com>
+ * @license http://opensource.org/licenses/MIT MIT
+ * @link https://packagist.org/packages/ramsey/uuid-doctrine Packagist
+ * @link https://github.com/ramsey/uuid-doctrine GitHub
+ */
+
+namespace Ramsey\Uuid\Doctrine;
+
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Id\AbstractIdGenerator;
+
+/**
+ * UUID generator for the Doctrine ORM.
+ */
+class UuidGenerator extends AbstractIdGenerator
+{
+    /**
+     * Generate an identifier
+     *
+     * @param \Doctrine\ORM\EntityManager  $em
+     * @param \Doctrine\ORM\Mapping\Entity $entity
+     * @return Ramsey\Uuid\Uuid
+     */
+    public function generate(EntityManager $em, $entity)
+    {
+        return \Ramsey\Uuid\Uuid::uuid4();
```

Comment:
The `CustomerIdGenerator` annotation only accepts a `class` option (see [here](https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/Mapping/CustomIdGenerator.php)) and the `GeneratedValue` annotation only accepts a `strategy` option (see [here](https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/Mapping/GeneratedValue.php)) so I can't see any way to inject configuration into the generator.

I could set up multiple Doctrine generator classes; one for each UUID version, though that still wouldn't allow using the different random generators. What do you think?